gdal/frmts/kea/keadataset.cpp

```diff
@@ -27,12 +27,245 @@
  *
  */
 
+#include <algorithm>
+#include <mutex>
 #include "keadataset.h"
 #include "keaband.h"
 #include "keacopy.h"
 
+extern "C" int CPL_DLL GDALIsInGlobalDestructor(void);
+
 CPL_CVSID("$Id$")
 
+/* adapted from hdf5dataset.cpp */
```

Comment:
Can't that be factored with the HDF5 implementation ? I don't like seeing this copy&paste

Reply:
Good point, but not sure how this can be achieved. We can't assume that hdf5 will be enabled, so can't access it. Probably not a good idea to have this put into the gdal core since it will be only valid when hdf5 headers available. 

Maybe we need some sort of hdf5 common code that is enabled when hdf5 or kea is enabled?

Reply:
> Maybe we need some sort of hdf5 common code that is enabled when hdf5 or kea is enabled?

A simple solution would be to put this in a .h file in frmts/hdf5. That would result in some binary duplication if both drivers are built, but not that much, so should be OK